plugins/inputs/openweathermap/openweathermap.go

```diff
@@ -0,0 +1,305 @@
+package openweathermap
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type OpenWeatherMap struct {
+	BaseUrl string
+	AppId   string
+	CityId  []string
+
+	client *http.Client
+
+	ResponseTimeout internal.Duration
+	Fetch           []string
+	Units           string
+}
+
+// https://openweathermap.org/current#severalid
+// Call for several city IDs
+// The limit of locations is 20.
+const owmRequestSeveralCityId int = 20
+const defaultResponseTimeout time.Duration = time.Second * 5
+const defaultUnits string = "metric"
+
+var sampleConfig = `
+  ## Root url of weather map REST API
+  base_url = "https://api.openweathermap.org/"
+  ## Your personal user token from openweathermap.org
+  app_id = "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
+  city_id = ["2988507", "2988588"]
+
+  ## HTTP response timeout (default: 5s)
+  response_timeout = "5s"
+  fetch = ["weather", "forecast"]
+  units = "metric"
+  ## Limit OpenWeatherMap query interval. See calls per minute info at: https://openweathermap.org/price
+  interval = "10m"
+`
+
+func (n *OpenWeatherMap) SampleConfig() string {
+	return sampleConfig
+}
+
+func (n *OpenWeatherMap) Description() string {
+	return "Read current weather and forecasts data from openweathermap.org"
+}
+
+func (n *OpenWeatherMap) Gather(acc telegraf.Accumulator) error {
+	var wg sync.WaitGroup
+	var strs []string
+
+	base, err := url.Parse(n.BaseUrl)
+	if err != nil {
+		return err
+	}
+
+	// Create an HTTP client that is re-used for each
+	// collection interval
+
+	if n.client == nil {
+		client, err := n.createHttpClient()
+		if err != nil {
+			return err
+		}
+		n.client = client
+	}
+	units := n.Units
+	if units == "" {
+		units = defaultUnits
+	}
+	for _, fetch := range n.Fetch {
+		if fetch == "forecast" {
+			var u *url.URL
+			var addr *url.URL
+
+			for _, city := range n.CityId {
+				u, err = url.Parse(fmt.Sprintf("/data/2.5/forecast?id=%s&APPID=%s&units=%s", city, n.AppId, units))
+				if err != nil {
+					acc.AddError(fmt.Errorf("Unable to parse address '%s': %s", u, err))
+					continue
+				}
+				addr = base.ResolveReference(u)
```

Comment:
Since this is always used with `.String()` change `addr` to a string and set `addr = base.ResolveReference(u).String()`, passing just the string into `gatherUrl`

same for line 106, 120, 122, etc...